fission/src/aps/APS.ts

```diff
@@ -116,14 +125,38 @@ class APS {
         }
     }
 
-    static async convertAuthToken(code: string, codeVerifier: string) {
+    static async refreshAuthToken(refresh_token: string) {
+        let now
+        if ((now = Date.now()) - lastCall > delay) {
+            lastCall = now
+            const res = await fetch('https://developer.api.autodesk.com/authentication/v2/token', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/x-www-form-urlencoded'
+                },
+                body: new URLSearchParams({
+                    'client_id': CLIENT_ID,
+                    'grant_type': 'refresh_token',
+                    'refresh_token': refresh_token,
+                    'scope': 'data:read',
+                })
+            }).then(res => res.json());
```

Comment:
Please check for error on refresh and handle accordingly. You should redirect the user to login again. In which case, you can also provide the email automatically as an autofill using [this](https://aps.autodesk.com/en/docs/oauth/v2/developers_guide/authoptions/). I did look into the refresh token, and the only thing I've found is a [stackoverflow post from 2017](https://stackoverflow.com/questions/43680794/does-the-refresh-token-expire/43681411#43681411) indicating they are "extremely long lived." Other than that, we just have to assume it can eventually expire.